src/containers/Tenant/Diagnostics/Partitions/Partitions.tsx

```diff
@@ -43,215 +40,162 @@ export const b = block('ydb-diagnostics-partitions');
+        dispatch(setSelectedConsumer(value));
     };
 
-    const handlePartitionIdSearchChange = (value: string) => {
-        setPartitionIdSearchValue(value);
-    };
+    const loading =
+        (nodesLoading && !nodesWasLoaded) ||
+        (consumerLoading && !consumerWasLoaded) ||
+        (topicLoading && !topicWasLoaded);
 
-    const handleGeneralSearchChange = (value: string) => {
-        setGeneralSearchValue(value);
-    };
+    const error = nodesError || topicError || consumerError;
 
-    if (error) {
-        return <ResponseError error={error} />;
-    }
+    const getContent = () => {
+        if (loading) {
+            return <TableSkeleton className={b('loader')} />;
+        }
+        if (error) {
+            return <ResponseError error={error} />;
+        }
 
-    if (!consumersToSelect || !consumersToSelect.length) {
-        return <div>{i18n(`noConsumersMessage.${isCdcStream ? 'stream' : 'topic'}`)}</div>;
-    }
+        return (
+            <DataTable
+                theme="yandex-cloud"
+                data={dataToRender}
+                columns={columnsToShow}
+                settings={DEFAULT_TABLE_SETTINGS}
+                emptyDataMessage={i18n('table.emptyDataMessage')}
+            />
+        );
+    };
 
     return (
         <div className={b()}>
-            <div className={b('controls')}>
-                <Select
-                    className={b('consumer-select')}
-                    placeholder={i18n('controls.consumerSelector.placeholder')}
-                    label={i18n('controls.consumerSelector')}
-                    options={consumersToSelect}
-                    value={[selectedConsumer || '']}
-                    onUpdate={handleConsumerSelectChange}
-                    filterable={consumers && consumers.length > 5}
-                />
-                <Search
-                    onChange={handlePartitionIdSearchChange}
-                    placeholder={i18n('controls.partitionSearch')}
-                    className={b('search', {partition: true})}
-                    value={partitionIdSearchValue}
-                />
-                <Search
-                    onChange={handleGeneralSearchChange}
-                    placeholder={i18n('controls.generalSearch')}
-                    className={b('search', {general: true})}
-                    value={generalSearchValue}
-                />
-                <TableColumnSetup
-                    key="TableColumnSetup"
-                    popupWidth="242px"
-                    items={columnsToSelect}
-                    showStatus
-                    onUpdate={hadleTableColumnsSetupChange}
-                    className={b('table-settings')}
-                />
-            </div>
+            <PartitionsControls
+                consumers={consumers}
+                selectedConsumer={selectedConsumer}
+                handleSelectedConsumerChange={handleSelectedConsumerChange}
```

Comment:
Just to reduce ambiguity - props should be called `onSomething`, its value - `handleSomething`